### PR TITLE
git subtree documentation

### PR DESCRIPTION
**Pull request by gchenfc**
Adds some information about the subtree relationship between gtsam and wrap, and also the command to update/merge/push wrap in GTSAM.

### CODE REVIEW

**Comment by ProfFan** [comment]
It's interesting that `./update_wrap.sh` alone does not do everything. @varunagrawal Did you have this problem?

**Review by dellaert — COMMENTED** [review]
I think the documentation about how wrap is included within GTSAM is supposef to go in GTSAM. Below I would be generic and cite GTSAM as an *example*

**Comment by varunagrawal** [comment]
I never had any problem with the update_wrap script. However, a failure during the running of the script can cause the branch clean up to not execute, compromising any future runs.

So unless Gerry had this issue and omitted the details, I'm not sure why we need that extra command at the end.

**Comment by gchenfc** [comment]
Hmm.  When I ran the ./update_wrap script, it updated it in my local gtsam folder, but then my git status / git diff was totally clean so I couldn't add/commit the updates.  Therefore I had to run that last command.

So for you guys you just run ./update_wrap then git commit git push?

**Comment by gchenfc** [comment]
@dellaert I initially did that but then thought it looked awkward in the python/README file so I moved it to wrap.  I'll find a better place for it in gtsam.

**Comment by dellaert** [comment]
Actually, maybe it's OK

**Comment by dellaert** [comment]
But at least (in your latest PR) make a link to this README file for "maintainers that need to update the wrapping code itself".
PS in python *and* MATLAB readme files.

**Comment by gchenfc** [comment]
TL;DR - Disregard the last line, actually everything works fine

Oops sorry I lied, I'm not sure why it wasn't working before (maybe as Varun mentioned I had some old error that I didn't notice before and that was now causing silent errors on all future runs?), but now the ./update_wrap.sh actually works exactly as expected.  I don't know what changed - before I wasn't getting any commits and my git was clean.  I suspect some odd order of switching branches and ./update_wrap sometimes causes git to think nothing changed but also the merge commits are on a different branch, but I'm having trouble reproducing it.

I'll update this documentation.

File: README.md [hunk 1]
```diff
@@ -109,3 +109,13 @@ Arguments:
   include_directories. Again, normally, leave this empty.
 - `extraMexFlags`: Any _additional_ flags to pass to the compiler when building
   the wrap code. Normally, leave this empty.
+
+## Git subtree and Contributing
+
+The [wrap library](https://github.com/borglab/wrap) is included in GTSAM as a git subtree.  This means that sometimes the wrap library can have new features or changes that are not yet reflected in GTSAM.  There are two options to get the most up-to-date versions of wrap:
+  1. Clone and install the [wrap repository](https://github.com/borglab/wrap).  For external projects, make sure cmake is using the external `wrap` rather than the one pre-packaged with GTSAM.
+  2. run `./update_wrap.sh` from the root of GTSAM's repository to pull in the newest version of wrap to your local GTSAM installation.  *Warning*: this creates 2 new commits in GTSAM.  Be sure to _not_ push these to master/develop.  Preferably, open up a new PR with these updates (see below).
```
Comment (varunagrawal) [inline README.md:117]: Capitalize run.

File: README.md [hunk 2]
```diff
@@ -109,3 +109,13 @@ Arguments:
...
+
+To make a PR on GTSAM with the most recent wrap updates, create a new branch/fork then pull in the most recent wrap changes using `./update_wrap.sh`.  You should find that two new commits have been made: a squash and a merge from master.  You can push these and open a PR.
```
Comment (varunagrawal) [inline README.md:119]: Add the warning right before this so that it stands out more?

**Review by varunagrawal — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (gchenfc) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (gchenfc) [reply README.md]: done

**Review by varunagrawal — APPROVED** [review]
LGTM
